could not broadcast input array from shape (32,256) into shape (32)
could not broadcast input array from shape (32,256) into shape (32)
could not broadcast input array from shape (32,256) into shape (32)
could not broadcast input array from shape (32,256) into shape (32)
could not broadcast input array from shape (32,256) into shape (32)
could not broadcast input array from shape (32,256) into shape (32)
could not broadcast input array from shape (32,256) into shape (32)
could not broadcast input array from shape (32,256) into shape (32)
could not broadcast input array from shape (32,256) into shape (32)
could not broadcast input array from shape (32,256) into shape (32)
could not broadcast input array from shape (32,256) into shape (32)
could not broadcast input array from shape (32,256) into shape (32)
could not broadcast input array from shape (32,256) into shape (32)
could not broadcast input array from shape (32,256) into shape (32)
could not broadcast input array from shape (32,256) into shape (32)
could not broadcast input array from shape (32,256) into shape (32)
could not broadcast input array from shape (32,256) into shape (32)
could not broadcast input array from shape (32,256) into shape (32)
could not broadcast input array from shape (32,256) into shape (32)
could not broadcast input array from shape (32,256) into shape (32)
could not broadcast input array from shape (32,256) into shape (32)
could not broadcast input array from shape (32,256) into shape (32)
could not broadcast input array from shape (32,256) into shape (32)
could not broadcast input array from shape (32,256) into shape (32)
could not broadcast input array from shape (32,256) into shape (32)
could not broadcast input array from shape (32,256) into shape (32)
could not broadcast input array from shape (32,256) into shape (32)
could not broadcast input array from shape (32,256) into shape (32)
could not broadcast input array from shape (32,256) into shape (32)
could not broadcast input array from shape (32,256) into shape (32)
could not broadcast input array from shape (32,512) into shape (32)
Saw a non-null label (index >= num_classes - 1) following a null label, batch: 0 num_classes: 5530 labels: 
	 [[Node: ctc/CTCLoss = CTCLoss[ctc_merge_repeated=true, ignore_longer_outputs_than_inputs=false, preprocess_collapse_repeated=false, _device="/job:localhost/replica:0/task:0/device:CPU:0"](ctc/Log, ctc/ToInt64, ctc/ToInt32_2, ctc/ToInt32_1)]]

Caused by op 'ctc/CTCLoss', defined at:
  File "train.py", line 141, in <module>
    main(argv=None)
  File "train.py", line 111, in main
    model, basemodel = om.get_model(aligned_height, nclass)
  File "D:\python_work\machine-learning\engineering\Ali-ICPR-proj\ocr\model.py", line 76, in get_model
    loss_out = Lambda(ctc_lambda_func, output_shape=(1,), name='ctc')([y_pred, labels, input_length, label_length])
  File "C:\Users\Administrator\AppData\Local\Programs\Python\Python35\lib\site-packages\keras\engine\topology.py", line 603, in __call__
    output = self.call(inputs, **kwargs)
  File "C:\Users\Administrator\AppData\Local\Programs\Python\Python35\lib\site-packages\keras\layers\core.py", line 651, in call
    return self.function(inputs, **arguments)
  File "D:\python_work\machine-learning\engineering\Ali-ICPR-proj\ocr\model.py", line 29, in ctc_lambda_func
    return K.ctc_batch_cost(labels, y_pred, input_length, label_length)
  File "C:\Users\Administrator\AppData\Local\Programs\Python\Python35\lib\site-packages\keras\backend\tensorflow_backend.py", line 3755, in ctc_batch_cost
    sequence_length=input_length), 1)
  File "C:\Users\Administrator\AppData\Local\Programs\Python\Python35\lib\site-packages\tensorflow\python\ops\ctc_ops.py", line 152, in ctc_loss
    ignore_longer_outputs_than_inputs=ignore_longer_outputs_than_inputs)
  File "C:\Users\Administrator\AppData\Local\Programs\Python\Python35\lib\site-packages\tensorflow\python\ops\gen_ctc_ops.py", line 222, in _ctc_loss
    name=name)
  File "C:\Users\Administrator\AppData\Local\Programs\Python\Python35\lib\site-packages\tensorflow\python\framework\op_def_library.py", line 787, in _apply_op_helper
    op_def=op_def)
  File "C:\Users\Administrator\AppData\Local\Programs\Python\Python35\lib\site-packages\tensorflow\python\framework\ops.py", line 2956, in create_op
    op_def=op_def)
  File "C:\Users\Administrator\AppData\Local\Programs\Python\Python35\lib\site-packages\tensorflow\python\framework\ops.py", line 1470, in __init__
    self._traceback = self._graph._extract_stack()  # pylint: disable=protected-access

InvalidArgumentError (see above for traceback): Saw a non-null label (index >= num_classes - 1) following a null label, batch: 0 num_classes: 5530 labels: 
	 [[Node: ctc/CTCLoss = CTCLoss[ctc_merge_repeated=true, ignore_longer_outputs_than_inputs=false, preprocess_collapse_repeated=false, _device="/job:localhost/replica:0/task:0/device:CPU:0"](ctc/Log, ctc/ToInt64, ctc/ToInt32_2, ctc/ToInt32_1)]]

